honey-badger/src/provider.rs

```diff
@@ -113,6 +139,46 @@ impl AuthProvider {
         Ok(())
     }
 
+    pub fn get_terms_and_conditions_status(
+        &self,
+        access_token: String,
+        terms: TermsAndConditions,
+    ) -> Result<TermsAndConditionsStatus> {
+        info!("Requesting T&C status ({terms:?})...");
+        ensure!(
+            self.auth_level == AuthLevel::Pseudonymous,
+            invalid_input(
+                "Requesting T&C status not supported for auth levels other than Pseudonymous"
+            )
+        );
+
+        let variables = get_terms_and_conditions_status::Variables {
+            service_provider: terms.into(),
+        };
+        let client = build_client(Some(&access_token))?;
+        let data =
+            post_blocking::<GetTermsAndConditionsStatus>(&client, &self.backend_url, variables)?;
+
+        let terms_status = data.get_terms_conditions_status.ok_or_runtime_error(
+            GraphQlRuntimeErrorCode::RemoteServiceUnavailable,
+            "Couldn't fetch T&C status.",
+        )?;
+
+        let accepted_at = if terms_status.accepted_terms {
+            terms_status
+                .accept_date
+                .map(|date| parse_from_rfc3339(&date))
+                .transpose()?
+        } else {
+            None
+        };
+
+        Ok(TermsAndConditionsStatus {
+            accepted_at,
+            terms_and_conditions: terms_status.service_provider.try_into()?,
```

Comment:
The service can return a `ServiceProviderEnum` other than the one we requested? If so, that would be a bug, right?

Instead of returning the terms and conditions provided by the service, it would be clearer if we check if they are the same we requested and if not, return an error.